pkg/apis/eventing/v1/trigger_types.go

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2020 The Knative Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"knative.dev/pkg/apis"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+	"knative.dev/pkg/kmeta"
+)
+
+const (
+	// DependencyAnnotation is the annotation key used to mark the sources that the Trigger depends on.
+	// This will be used when the kn client creates a source and trigger pair for the user such that the trigger only receives events produced by the paired source.
+	DependencyAnnotation = "knative.dev/dependency"
+	// InjectionAnnotation is the annotation key used to enable knative eventing injection for a namespace and automatically create a default broker.
+	// This will be used when the client creates a trigger paired with default broker and the default broker doesn't exist in the namespace
+	InjectionAnnotation = "knative-eventing-injection"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
```

Comment:
```suggestion
// +genclient
// +genreconciler:krshapedlogic=true
// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
```

v1beta1 api has `// +genreconciler:krshapedlogic=true`. Why we don't need it here?

Reply:
Ah, it's only cause I didn't yet use it. Changed 👍